rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -852,6 +878,36 @@ int SqliteStorage::read_db_schema_version()
   return schema_version;
 }
 
+uint64_t SqliteStorage::read_total_page_count_locked()
+{
+  auto scope_exit_reset_statement = rcpputils::make_scope_exit(
+    [this]() {
+      page_count_statement_->reset();
+    });
+  auto page_count_query_result = page_count_statement_->execute_query<int>();
+  auto page_count_query_result_begin = page_count_query_result.begin();
```

Comment:
This is just a matter of taste, but it seems unnecessary to assign a variable here.  You are only saving 2 characters while typing :).

Reply:
Unfortunately not. And in this case this sort of workaround with temporary `page_count_query_result_begin` is required because when we call `page_count_query_result.begin()` we actually calling `statement_->step()` inside and check if we have already called begin() iterator to not allow to call it more than once.
```
    Iterator begin()
    {
      try_access_data();
      return Iterator(statement_, 0);
    }
    void try_access_data()
    {
      if (is_already_accessed_) {
        throw SqliteException("Only one iterator per query result is supported!");
      }
      is_already_accessed_ = true;
    }
```
The rationale for caching `query_result.begin()` is because we need to call it twice (one more time in the return statement at the end of the function) and API allows to call it only once.
One more "skeleton in the SQLite3 wrapper API cabinet"